syncserver-db-common/src/lib.rs

```diff
@@ -97,212 +38,33 @@ impl From<deadpool::Status> for PoolState {
-        bso: Option<String>,
-    ) -> DbFuture<'_, SyncTimestamp> {
-        // If there's no collection, we return the overall storage timestamp
-        let collection = match collection {
-            Some(collection) => collection,
-            None => return Box::pin(self.get_storage_timestamp(user_id)),
-        };
-        // If there's no bso, return the collection
-        let bso = match bso {
-            Some(bso) => bso,
-            None => {
-                return Box::pin(
-                    self.get_collection_timestamp(params::GetCollectionTimestamp {
-                        user_id,
-                        collection,
-                    })
-                    .or_else(|e| {
-                        if e.is_collection_not_found() {
-                            future::ok(SyncTimestamp::from_seconds(0f64))
-                        } else {
-                            future::err(e)
-                        }
-                    }),
-                )
-            }
-        };
-        Box::pin(
-            self.get_bso_timestamp(params::GetBsoTimestamp {
-                user_id,
-                collection,
-                id: bso,
-            })
-            .or_else(|e| {
-                if e.is_collection_not_found() {
-                    future::ok(SyncTimestamp::from_seconds(0f64))
-                } else {
-                    future::err(e)
-                }
-            }),
-        )
-    }
-
-    /// Internal methods used by the db tests
-
-    fn get_collection_id(&self, name: String) -> DbFuture<'_, i32>;
-
-    fn create_collection(&self, name: String) -> DbFuture<'_, i32>;
-
-    fn update_collection(&self, params: params::UpdateCollection) -> DbFuture<'_, SyncTimestamp>;
-
-    fn timestamp(&self) -> SyncTimestamp;
-
-    fn set_timestamp(&self, timestamp: SyncTimestamp);
-
-    fn delete_batch(&self, params: params::DeleteBatch) -> DbFuture<'_, ()>;
-
-    fn clear_coll_cache(&self) -> DbFuture<'_, ()>;
-
-    fn set_quota(&mut self, enabled: bool, limit: usize, enforce: bool);
-}
-
-impl<'a> Clone for Box<dyn Db<'a>> {
-    fn clone(&self) -> Box<dyn Db<'a>> {
-        self.box_clone()
-    }
-}
-
-#[derive(Debug, Deserialize, Clone, Eq, PartialEq, Copy)]
-#[serde(rename_all = "lowercase")]
-pub enum Sorting {
-    None,
-    Newest,
-    Oldest,
-    Index,
-}
-
-impl Default for Sorting {
-    fn default() -> Self {
-        Sorting::None
-    }
-}
-
-#[derive(Clone, Debug, Default, Eq, Hash, PartialEq)]
-pub struct UserIdentifier {
-    /// For MySQL database backends as the primary key
-    pub legacy_id: u64,
-    /// For NoSQL database backends that require randomly distributed primary keys
-    pub fxa_uid: String,
-    pub fxa_kid: String,
+pub async fn run_on_blocking_threadpool<F, T, E, M>(f: F, e: M) -> Result<T, E>
+where
+    F: FnOnce() -> Result<T, E> + Send + 'static,
+    T: Send + 'static,
+    E: fmt::Debug + Send + 'static,
+    M: FnOnce(String) -> E,
+{
+    web::block(f)
```

Comment:
Unfortunately, that we have to use Actix's blocking threadpool instead of tokio's forces actix web to be a dependency of this crate